slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -65,12 +65,14 @@ trait PostgresProfile extends JdbcProfile {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       val VarCharPattern = "^'(.*)'::character varying$".r
       val IntPattern = "^\\((-?[0-9]*)\\)$".r
+      val DoublePattern = "[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?".r
```

Comment:
Unlike in the IntPattern, which captures the sign and all the digits, here you capture only the exponential part `([eE][-+]?[0-9]+)`